shared/actions/chat/index.js

```diff
@@ -923,64 +876,45 @@ function * _openConversation ({payload: {conversationIDKey}}: Constants.OpenConv
-    safeTakeEvery('chat:openAttachmentPopup', Attachment.onOpenAttachmentPopup),
-    safeTakeLatest('chat:openFolder', _openFolder),
-    safeTakeLatest('chat:badgeAppForChat', _badgeAppForChat),
-    safeTakeEvery(changedFocus, _changedFocus),
-    safeTakeEvery('chat:deleteMessage', Messages.deleteMessage),
-    safeTakeEvery('chat:openTlfInChat', _openTlfInChat),
-    safeTakeEvery('chat:loadedInbox', _ensureValidSelectedChat, true, false),
-    safeTakeEvery('chat:updateInboxComplete', _ensureValidSelectedChat, false, false),
-    safeTakeEvery('chat:saveAttachmentNative', Attachment.onSaveAttachmentNative),
-    safeTakeEvery('chat:shareAttachment', Attachment.onShareAttachment),
+    Saga.safeTakeSerially('chat:loadInbox', Inbox.onLoadInboxMaybeOnce),
+    Saga.safeTakeLatest('chat:inboxStale', Inbox.onLoadInbox),
+    Saga.safeTakeEvery('chat:loadMoreMessages', Saga.cancelWhen(_threadIsCleared, _loadMoreMessages)),
+    Saga.safeTakeLatest('chat:selectConversation', _selectConversation),
+    Saga.safeTakeEvery('chat:updateBadging', _updateBadging),
+    Saga.safeTakeEvery('chat:setupChatHandlers', _setupChatHandlers),
+    Saga.safeTakeEvery('chat:incomingMessage', _incomingMessage),
+    Saga.safeTakeEvery('chat:markThreadsStale', _markThreadsStale),
+    Saga.safeTakeEvery('chat:muteConversation', _muteConversation),
+    Saga.safeTakeEvery('chat:blockConversation', _blockConversation),
+    Saga.safeTakeEvery('chat:newChat', _newChat),
+    Saga.safeTakeEvery('chat:postMessage', Messages.postMessage),
+    Saga.safeTakeEvery('chat:editMessage', Messages.editMessage),
+    Saga.safeTakeEvery('chat:retryMessage', Messages.retryMessage),
+    Saga.safeTakeEvery('chat:startConversation', _startConversation),
+    Saga.safeTakeEvery('chat:updateMetadata', _updateMetadata),
+    Saga.safeTakeEvery('chat:appendMessages', _sendNotifications),
+    Saga.safeTakeEvery('chat:selectAttachment', Attachment.onSelectAttachment),
+    Saga.safeTakeEvery('chat:openConversation', _openConversation),
+    Saga.safeTakeEvery('chat:getInboxAndUnbox', Inbox.getInboxAndUnbox),
+    Saga.safeTakeEvery('chat:loadAttachment', Attachment.onLoadAttachment),
+    Saga.safeTakeEvery('chat:openAttachmentPopup', Attachment.onOpenAttachmentPopup),
+    Saga.safeTakeLatest('chat:openFolder', _openFolder),
+    Saga.safeTakeLatest('chat:badgeAppForChat', _badgeAppForChat),
+    Saga.safeTakeEvery(changedFocus, _changedFocus),
+    Saga.safeTakeEvery('chat:deleteMessage', Messages.deleteMessage),
+    Saga.safeTakeEvery('chat:openTlfInChat', _openTlfInChat),
+    Saga.safeTakeEvery('chat:loadedInbox', _ensureValidSelectedChat, true, false),
+    Saga.safeTakeEvery('chat:updateInboxComplete', _ensureValidSelectedChat, false, false),
+    Saga.safeTakeEvery('chat:saveAttachmentNative', Attachment.onSaveAttachmentNative),
+    Saga.safeTakeEvery('chat:shareAttachment', Attachment.onShareAttachment),
   ]
 }
 
 export default chatSaga
 
 export {
-  addPending,
```

Comment:
Only passing through creators to externally used actions. Chat internal stuff can know about ./creators